ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -0,0 +1,146 @@
+
+  private def tupleErrorsToResponse(errorTuple: (List[SchemaDeletionErrors], List[KafkaDeletionErrors])): List[DeletionEndpointResponse] = {
+    schemaErrorsToResponse(errorTuple._1) ::: kafkaErrorsToResponse(errorTuple._2)
+  }
+
+  private def schemaErrorsToResponse(schemaResults: List[SchemaDeletionErrors]): List[DeletionEndpointResponse] = {
+    schemaResults.flatMap(_.schemaDeleteTopicErrorList.errors.map(e => DeletionEndpointResponse(e.getSubject, e.errorMessage)).toList)
+  }
+
+  private def kafkaErrorsToResponse(kafkaResults: List[KafkaDeletionErrors]): List[DeletionEndpointResponse] = {
+    kafkaResults.flatMap(_.kafkaDeleteTopicErrorList.errors.map(e => DeletionEndpointResponse(e.topicName, e.errorMessage)).toList)
+  }
+
+  final case class DeletionRequest(topics: List[String])
+  implicit val deleteRequestFormat: RootJsonFormat[DeletionRequest] = jsonFormat1(DeletionRequest)
+
+  private def validResponse(topics: List[String], userDeleting: String)(implicit ec: ExecutionContext) = {
+    topics.map(topicName => addPromHttpMetric(topicName, StatusCodes.OK.toString(), "/deleteTopics/"))
+    topics.map(topicName => log.info(s"User $userDeleting deleted topic: $topicName"))
+    complete(StatusCodes.OK, topics)
+  }
+
+  private def returnResponse(topics: List[String], userDeleting: String, response: List[DeletionEndpointResponse])(implicit ec: ExecutionContext) = {
+    if (response.length == topics.length) {
+      topics.map(topicName => addPromHttpMetric(topicName, StatusCodes.InternalServerError.toString(), "/deleteTopics"))
+      complete(StatusCodes.InternalServerError, response)
+    } else {
+      val failedTopicNames = response.map(der => der.topicOrSubject)
+      val successfulTopics = topics.toSet.diff(failedTopicNames.toSet).toList
+      failedTopicNames.map(topicName => addPromHttpMetric(topicName, StatusCodes.Accepted.toString(), "/deleteTopics/"))
+      successfulTopics.map(topicName => addPromHttpMetric(topicName, StatusCodes.OK.toString(), "/deleteTopics/"))
+      successfulTopics.map(topicName => log.info(s"User $userDeleting deleted topic $topicName"))
+      complete(StatusCodes.Accepted, response)
+    }
+  }
+
+  private def invalidResponse(topics: List[String], userDeleting: String, e: NonEmptyList[DeleteTopicError])(implicit ec: ExecutionContext) = {
+    val allErrors = e.foldLeft((List.empty[SchemaDeletionErrors], List.empty[KafkaDeletionErrors])) { (agg, i) =>
+      i match {
+        case sde: SchemaDeletionErrors => (agg._1 :+ sde, agg._2)
+        case kde: KafkaDeletionErrors => (agg._1, agg._2 :+ kde)
+      }
+    }
+    val response = tupleErrorsToResponse(allErrors)
+    returnResponse(topics, userDeleting, response)
+  }
+
+  private def deleteTopics(topics: List[String], userDeleting: String)(implicit ec: ExecutionContext) = {
+    onComplete(
+      Futurable[F].unsafeToFuture(deletionProgram.deleteTopic(topics))
+    ) {
+      case Success(maybeSuccess) => {
```

Comment:
This is just personal preference, so take it or leave it, but I usually don't put the extra `{` because I think it makes it harder to read in many cases. Your call though.